cfgov/unprocessed/js/organisms/MegaMenu.js

```diff
@@ -0,0 +1,225 @@
+  var _flyoutMenu = new FlyoutMenu( _dom,
+                                    '.' + BASE_CLASS + '_trigger',
+                                    '.' + BASE_CLASS + '_content' ).init();
+  var _activeMenu = _flyoutMenu;
+  var _activeMenuDom = _flyoutMenu.getDom().content;
+  var _menuItems = _dom.querySelectorAll( '.' + BASE_CLASS + '_content-item' );
+  var _subMenus = {};
+
+  // TODO: Move tab trigger to its own class.
+  var _tabTriggerDom = _dom.querySelector( '.' + BASE_CLASS + '_tab-trigger' );
+
+  var KEY_TAB = 9;
+
+  /**
+   * @returns {Object} The MegaMenu instance.
+   */
+  function init() {
+    var initEventsBinded = _initEvents.bind( this );
+    var menuItem;
+    var submenu;
+    for ( var i = 1, len = _menuItems.length; i < len; i++ ) {
+      menuItem = _menuItems[i];
+      if ( menuItem.querySelector( '.u-link__disabled' ) === null ) {
+        submenu =
+          new FlyoutMenu( menuItem, '.' + BASE_CLASS + '_content-link',
+                          '.' + BASE_CLASS + '_subcontent',
+                          '.' + BASE_CLASS + '_subcontent-btn__back' ).init();
+        _subMenus[menuItem] = submenu;
+        initEventsBinded( submenu );
+      }
+    }
+
+    initEventsBinded( _flyoutMenu );
+    _tabTriggerDom.addEventListener( 'keyup', _handleTabPress );
+
+    return this;
+  }
+
+  /**
+   * @param {FlyoutMenu} menu - The FlyoutMenu to add event listeners to.
+   */
+  function _initEvents( menu ) {
+    menu.addEventListener( 'toggle', _handleToggle.bind( this ) );
+    menu.addEventListener( 'expandBegin', _handleExpandBegin );
+    menu.addEventListener( 'collapseBegin', _handleCollapseBegin );
+  }
+
+  /**
+   * Event handler for when there's a click on the page's body.
+   * Used to close the global search, if needed.
+   * @param {MouseEvent} event The event object for the mousedown event.
+   */
+  function _handleBodyClick( event ) {
+    var target = event.target;
+
+    var isInDesktop = _isInDesktop();
+    if ( isInDesktop && !_isDesktopTarget( target ) ||
+         !isInDesktop && !_isMobileTarget( target ) ) {
+      collapse();
+    }
+  }
+
+  // TODO: Move this to breakpoint-state.js.
+  /**
+   * Whether currently in the desktop view.
+   * @returns {boolean} True if in the desktop view, otherwise false.
+   */
+  function _isInDesktop() {
+    var isInDesktop = false;
+    var currentBreakpoint = breakpointState.get();
+    if ( currentBreakpoint.isBpMED ||
+         currentBreakpoint.isBpLG ||
+         currentBreakpoint.isBpXL ) {
+      isInDesktop = true;
+    }
+    return isInDesktop;
+  }
+
+  /**
+   * Whether a target is one of the ones that appear in the desktop view.
+   * @param {HTMLNode} target - The target of a mouse event (most likely).
+   * @returns {boolean} True if the passed target is in the desktop view.
+   */
+  function _isDesktopTarget( target ) {
+    var bodyElem = document.body;
+    target = target.parentNode;
+    do {
+      if ( target === _activeMenuDom ) {
```

Comment:
Moved to dom-traverse.js in https://github.com/cfpb/cfgov-refresh/commit/072a94a59637ebbb3efd575ee220760a02f98341